code/ai/aigoals.cpp

```diff
@@ -134,28 +136,41 @@ const char *Ai_goal_text(int goal)
 	return NULL;
 }
 
-// function to maybe add the form on my wing goal for a player's starting wing.  Called when a player wing arrives.
-void ai_maybe_add_form_goal( wing *wingp )
+void ai_maybe_add_form_goal(wing* wingp)
 {
 	int j;
 
 	// iterate through the ship_index list of this wing and check for orders.  We will do
 	// this for all ships in the wing instead of on a wing only basis in case some ships
 	// in the wing actually have different orders than others
-	for ( j = 0; j < wingp->current_count; j++ ) {
-		ai_info *aip;
+	for (j = 0; j < wingp->current_count; j++) {
+		ai_info* aip;
 
-		Assert( wingp->ship_index[j] != -1 );						// get Allender
+		Assert(wingp->ship_index[j] != -1);						// get Allender
 
 		aip = &Ai_info[Ships[wingp->ship_index[j]].ai_index];
 		// don't process Player_ship
-		if ( aip == Player_ai )
+		if (aip == Player_ai || (MULTIPLAYER_MASTER && Objects[Ships[aip->shipnum].objnum].flags[Object::Object_Flags::Player_ship])) {
 			continue;
-		
+		}
+
 		// it is sufficient enough to check the first goal entry to see if it has a valid goal
-		if ( aip->goals[0].ai_mode == AI_GOAL_NONE ) {
-			// need to add a form on my wing goal here.  Ships are always forming on the player's wing.
-			ai_add_ship_goal_player( AIG_TYPE_PLAYER_SHIP, AI_GOAL_FORM_ON_WING, -1, Player_ship->ship_name, aip );
+		if (aip->goals[0].ai_mode == AI_GOAL_NONE) {
+			// Need to have a more specific target in multi, or they may end up trying to target standalone placeholder.
+			// So let's have them target their team leader.  In dogfight, all player slot ai die, so just exclude.
+			if (MULTIPLAYER_MASTER && !(Netgame.type_flags & NG_TYPE_DOGFIGHT)) {
+
+				if (Netgame.type_flags & NG_TYPE_TEAM) {
+					const ship_registry_entry* ship_regp = ship_registry_get(Ships[wingp->ship_index[j]].ship_name);
+					ai_add_ship_goal_player(AIG_TYPE_PLAYER_SHIP, AI_GOAL_FORM_ON_WING, -1, Ships[Wings[TVT_wings[ship_regp->p_objp->team]].special_ship].ship_name, aip);
+				}
+				else {
+					ai_add_ship_goal_player(AIG_TYPE_PLAYER_SHIP, AI_GOAL_FORM_ON_WING, -1, Ships[Wings[Starting_wings[0]].special_ship].ship_name, aip);
```

Comment:
The `special_ship` field refers to the position in the wing, not the ship's shipnum.  But regardless, I am about to remove the `special_ship` field, because it has a fundamental design flaw.  Instead, have the ships form on the wing's ship_index[0]:
```
Ships[Wings[TVT_wings[ship_regp->p_objp->team]].ship_index[0]].ship_name

Ships[Wings[Starting_wings[0]].ship_index[0]].ship_name
```

Also, as an edge case, if for some reason the wing leader is AI-controlled, it will be given an order to form on itself.  That will need to be checked.

Reply:
Will do

Reply:
As it turns out, `special_ship` might be fixable, so do this:
```
int wingnum = TVT_wings[ship_regp->p_objp->team];
...Ships[Wings[tvt_wingnum].ship_index[Wings[tvt_wingnum].special_ship]].ship_name...

int wingnum = Starting_wings[0];
...Ships[Wings[wingnum].ship_index[Wings[wingnum].special_ship]].ship_name...
```

Reply:
All right. Thanks for putting that together!